dip-0032.md

```diff
@@ -0,0 +1,366 @@
+
+| Value | Size | Format | Example |
+| ----- | ---- | ------ | ------- |
+| < `0xFD` | 1 byte | `uint8_t` | `0x0F` = 15 |
+| <= `0xFFFF` | 3 bytes | `0xFD` followed by the number as a `uint16_t` | `0xFD 00FF` = 65 280 |
+| <= `0xFFFF FFFF` | 5 bytes | `0xFE` followed by the number as a `uint32_t` | `0xFE 0000 00FF` = 4 278 190 080 |
+| <= `0xFFFF FFFF FFFF FFFF` | 9 bytes | `0xFF` followed by the number as a `uint64_t` | `0xFF 0000 0000 0000 00FF` = 18 374 686 479 671 623 680 |
+
+#### Vector\<Type\>
+
+Each `Vector` begins with a `VarInt` describing the number of items it contains.
+
+If the vector is of type `hex`, then the size / structure of each individual item is not known in advance. In this case, each item begins with a `VarInt` describing its size `s` in bytes, followed by `s` bytes which should be interpreted as the item itself.
+Otherwise, size prefixes are omitted, and each item should be interpreted in accordance with the vector's type.
+
+In other words, the vector is serialized as follows: `[Length (n)][Item #1][Item #2][...][Item #n]`.
+
+#### extra_payload
+
+The structure of `extra_payload` is :
+
+| Field | Type | Size | Description |
+| ----- | ---- | ---- | ----------- |
+|numTxInputs| VarInt | varies | Number of confidential inputs|
+| txInputs| `Vector<TxInput>`| 33*numTxInputs bytes| Confidential transaction inputs |
+|numTxOutputs| VarInt | varies | Number of confidential outputs |
+| txOutputs | `Vector<TxOutput>`| 33*numTxOutputs bytes| Confidential transaction outputs |
+| proof |ConfidentialProof| Varies | Confidential proof data |
+
+# TxInput
+
+| Field | Required | Size | Data Type | Encoding | Notes |
+| ----- | -------- | ---- | --------- | -------- | ----- |
+| txid | Yes | 32 bytes | `hex` | See NOTE below | |
+| outIndex | Yes | 4 bytes | `uint32_t` | Little-endian | **Input is a coinbase**: `0xffffffff` |
+| scriptSig Length | Yes | Varies | `VarInt` | |  |
+| ScriptSig | Yes  | Varies | `hex` | | |
+| Sequence | Yes | 4 bytes | `uint32_t` | Little-endian | |
+
+NOTE: The hex encodings of hashes are byte-reversed, and so the bytes will need to be re-reversed to match the serialized data. This is the same situation as in Bitcoin. For example, the hash `1123...deff` would be displayed by the Bitcoin and Dash clients as `ffde...2311`. This is primarily for historical reasons: the Bitcoin and Dash clients have always interpreted and displayed hashes as little-endian integers and parsed their bytes in reverse order.
+
+# TxOutput
+
+| Field | Type | Size | Description |
+| ----- | ---- | ---- | ----------- |
+| amount|ConfidentialAmount| 33 bytes| The confidential amount being transferred |
+| nonce |ConfidentialNonce| 33 bytes| The confidential nonce |
+| proof |ConfidentialProof| Varies | The rangeproof that the confidential amount being transferred is valid |
+
+The following data structures have been adapted from the Elements Project Transaction format:
+
+#### ConfidentialAmount
```

Comment:
See the lint message here about heading levels. This one (and one or more of the following) are at the incorrect level. Could you update them to so we aren't skipping levels?